### PR TITLE
[Violations] Update tagOutOfPolicy and taxOutOfPolicy 

### PR DESCRIPTION
**Pull request by neonbhai**
### Details
We update the violation messages for `tagOutOfPolicy` and `taxOutOfPolicy`

### Fixed Issues

$ https://github.com/Expensify/App/issues/34539
PROPOSAL: https://github.com/Expensify/App/issues/34539#issuecomment-1892817935

### Tests

<Details><Summary>Prerequisite:</Summary>

1. Create a collect policy in OldDot
2. Execute the following JS snippet in the OldDot policy to enable workspace chats creation:
    ```js
    if (Url.getCurrentPageName()!="policy" ){ alert('Only run this snippet from a policy editor!'); } else { var p = Policy.getCurrent(); p.policy.isPolicyExpenseChatEnabled = "true"; p.save().done(function(){$.jGrowl('Workspace chat creation enabled!')}); }
     ```
3. Add an employee to the policy
4. Enable Tags and add a custom Tag. 
</Details>

#### For Tag Out Of Policy:
1. Log in to the employee's account
2. Tap on the green plus button (FAB)
3. Select Request money
4. Select any currency and amount
5. Click next
6. Choose the WS room
7. Click on "Show more"
8. Choose the custom tag in the Tag section
9. Complete the IOU
10. Login admin account on OldDot (expensify.com)
11. Navigate to the Collect group policy settings
12. In the Tag tab, disable the employee's selected tag from step `8`
13. In the employee's account in ND, open the IOU details
14. Verify the Tag violation can be seen as: `Tag no longer valid`

- [x] Verify that no errors appear in the JS console

#### For Tax Out Of Policy:
We cannot test the `taxOutOfPolicy` violation as we do not yet render it in [MoneyRequestView](https://github.com/Expensify/App/blob/a1fe671e94f69a694e2847a6b74685270149e991/src/components/ReportActionItem/MoneyRequestView.js#L119).

### Offline tests
Violations for a new iou will not be received when offline as they are sent from the backend.

### QA Steps

Same as Tests step.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>
Screenshot 2024-02-06 at 3 21 35 AM

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot 2024-02-05 at 9 56 31 PM

</details>

<details>
<summary>iOS: Native</summary>
ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Screenshot 2024-02-05 at 9 37 42 PM

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-01-24 at 1 10 05 PM

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-02-06 at 12 22 43 AM

</details>

### CODE REVIEW

File: src/languages/en.ts
Comment (s77rt) [inline src/languages/en.ts:1]: Can you please do the same for `categoryOutOfPolicy` and `customUnitOutOfPolicy` for consistency

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/languages/en.ts
Comment (cead22) [reply src/languages/en.ts:1]: We don't need to do this for either of those

**Review by cead22 — DISMISSED** [review]

**Comment by cead22** [comment]
Please update tests to include both cases: the single level tags, and the multi level tags. To setup multi level tags you can follow these instructions https://community.expensify.com/discussion/5756/how-to-set-up-and-manage-multi-level-tagging?utm_source=community-search&utm_medium=organic-search&utm_term=independent+tag, and use this file 
[multitags.csv](https://github.com/Expensify/App/files/14058274/multitags.csv)


That file will give the tag lists a name, which should produce violations like "Missing Project"

File: src/languages/en.ts
Comment (cead22) [reply src/languages/en.ts:1]: For context, check this out https://github.com/Expensify/App/issues/34539#issuecomment-1894303047. That only applies to tags, but not categories or custom units, so "Category no longer valid" and "Unit no longer valid" are good copies for these violations

**Comment by s77rt** [comment]
Code looks good but I'm unable to test as I'm not getting any violations errors. I asked for help on [Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1706318605848799)

**Comment by s77rt** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


android

</details>

<details>
<summary>Android: mWeb Chrome</summary>


mweb-chrome

</details>

<details>
<summary>iOS: Native</summary>


ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>


mweb-safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


web

</details>

<details>
<summary>MacOS: Desktop</summary>


desktop

</details>

**Comment by s77rt** [comment]
@neonbhai Please tag me once the PR is updated (added new testing steps and missing screenshots, complete the checklist)

**Comment by s77rt** [comment]
@neonbhai Any updates here?

**Comment by cead22** [comment]
Bump @neonbhai! I also pinged you on #expensify-open-source

**Comment by cead22** [comment]
> * Am I missing something? 🤔

No, it's just that multiple tags isn't currently supported in new dot, so what you're seeing is what's expected. We also have to update some API commands so when you update a request, violations clear automatically (which is something I'm working on)

> * Looks like backend still doesn't send violations with tagName

I think I have an issue for this as well -- but I'll look into it, but let's not block this PR on that

**Comment by s77rt** [comment]
@neonbhai Can you raise the translation question in Slack?

**Comment by cead22** [comment]
The translations are good, let's not block on this

**Comment by s77rt** [comment]
@neonbhai Please apply the same fix for the Spanish translation and we should be good to merge

File: src/languages/es.ts [hunk 1]
```diff
@@ -2641,7 +2641,7 @@ export default {
         someTagLevelsRequired: 'Falta etiqueta',
         tagOutOfPolicy: ({tagName}: ViolationsTagOutOfPolicyParams) => `La etiqueta ${tagName} ya no es válida`,
```
Comment (s77rt) [inline src/languages/es.ts:2642]: This is not fixed yet

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by cead22** [comment]
> Is this good?

Depends, can you share the policy setup, to see if it has multiple levels of tags?

File: src/languages/es.ts (hunk 1 above)
Comment (cead22) [reply src/languages/es.ts:2642]: I'll fix this on the back end so we can rely on `tagName` being sent to the client, without having to check if `tagName === 'Tag'

**Comment by cead22** [comment]
Yeah that looks good, and I submitted a back end fix so when the tagName is `Tag` we don't send it. Is there anything else that looks like it's not working as expected, or does everything work fine now?

**Comment by s77rt** [comment]
@neonbhai In step 12 you are referring to the wrong step (should be 8 not 10)

**Comment by s77rt** [comment]
> 16. Verify the Tag violation can be seen as: `La etiqueta ya no es válida`

I don't think that's the case yet, still getting `La etiqueta Tag ya no es válida` Looks like the BE change are not live yet. If you are getting same thing just omit that test step

**Review by s77rt — APPROVED** [review]

**Review by luacmartins — APPROVED** [review]

**Comment by luacmartins** [comment]
@cead22 all yours!

**Review by cead22 — APPROVED** [review]

**Pull request merged**